src/Server.py

```diff
@@ -357,13 +362,13 @@ def search(self, search_term: str) -> list:
         return res[:self.num_res_return]
 
     @staticmethod
-    def __search_helper(word_list, idx, path, res):
-        if idx == len(word_list):
+    def __search_helper(words: Word_list, idx: int, path: Word_list, res: Word_lists):
+        if idx == len(words):
             res.append(list(path))
             return
-        for word in word_list[idx]:
+        for word in words[idx]:
             path.append(word)
-            Server.__search_helper(word_list, idx + 1, path, res)
+            Server.__search_helper(words, idx + 1, path, res)
```

Comment:
I think the only reason you change `word_list` to `words` is as `word_list` became a type, otherwise `words` and `word_list` are exactly the same (list if strings), rite ?

Reply:
Correct, the IDE hints that using the old name shadows the type we defined outside the function

Reply:
LGTM